src/model_theory/fraisse.lean

```diff
@@ -142,5 +162,82 @@ begin
     exact ⟨PM.equiv_range.symm⟩ }
 end
 
+/-- The age of a direct limit of structures is the union of the ages of the structures. -/
+@[simp] theorem age_direct_limit {ι : Type w} [preorder ι] [is_directed ι (≤)] [nonempty ι]
+  (G : ι → Type (max w w')) [Π i, L.Structure (G i)]
+  (f : Π i j, i ≤ j → G i ↪[L] G j) [directed_system G (λ i j h, f i j h)] :
+  L.age (direct_limit G f) = ⋃ (i : ι), L.age (G i) :=
+begin
+  classical,
+  ext M,
+  simp only [mem_Union],
+  split,
+  { rintro ⟨Mfg, ⟨e⟩⟩,
+    obtain ⟨s, hs⟩ := Mfg.range e.to_hom,
+    let out := @quotient.out _ (direct_limit.setoid G f),
+    obtain ⟨i, hi⟩ := finset.exists_le (s.image (sigma.fst ∘ out)),
+    have e' := ((direct_limit.of L ι G f i).equiv_range.symm.to_embedding),
+    refine ⟨i, Mfg, ⟨e'.comp ((substructure.inclusion _).comp e.equiv_range.to_embedding)⟩⟩,
+    rw [← hs, closure_le],
+    intros x hx,
+    refine ⟨f (out x).1 i (hi (out x).1 (finset.mem_image_of_mem _ hx)) (out x).2, _⟩,
+    simp only [directed_system.coe_nat_le_rec, eq_mpr_eq_cast, cast_cast,
+      embedding.coe_to_hom, direct_limit.of_apply],
+    rw [quotient.mk_eq_iff_out, direct_limit.equiv_iff G f _
+      (hi (out x).1 (finset.mem_image_of_mem _ hx)), directed_system.map_self];
+    refl },
+  { rintro ⟨i, Mfg, ⟨e⟩⟩,
+    exact ⟨Mfg, ⟨embedding.comp (direct_limit.of L ι G f i) e⟩⟩ }
+end
+
+/-- Sufficient conditions for a class to be the age of a countably-generated structure. -/
+theorem exists_cg_is_age_of (hn : K.nonempty)
+  (h : ∀ (M N : bundled.{w} L.Structure), nonempty (M ≃[L] N) → (M ∈ K ↔ N ∈ K))
+  (hc : (quotient.mk '' K).countable)
+  (fg : ∀ (M : bundled.{w} L.Structure), M ∈ K → Structure.fg L M)
+  (hereditary : hereditary K)
+  (joint_embedding : joint_embedding K) :
+  ∃ (M : bundled.{w} L.Structure), Structure.cg L M ∧ L.age M = K :=
+begin
+  obtain ⟨F, hF⟩ := hc.exists_surjective (hn.image _),
+  simp only [set.ext_iff, forall_quotient_iff, mem_image, mem_range, quotient.eq] at hF,
+  simp_rw [quotient.eq_mk_iff_out] at hF,
+  have hF' : ∀ n : ℕ, (F n).out ∈ K,
+  { intro n,
+    obtain ⟨P, hP1, hP2⟩ := (hF (F n).out).2 ⟨n, setoid.refl _⟩,
+    exact (h _ _ hP2).1 hP1 },
+  have hj := λ (N : K) (n : ℕ), joint_embedding N N.2 (F (n + 1)).out (hF' _),
```

Comment:
Maybe adding a line `choose M hMK hF hN using hj,` (or similar) would clean up the rest of the proof a bit.